### ISSUE TITLE
[Bug] package.json script with ${CI} causes "Error: Unbound variable"

### ISSUE THREAD

**Issue by bradleyayers** [yarnpkg/berry#223]
**Describe the bug**

It's not possible to use an environment variable in a `package.json` script. e.g.

```json
{
  "scripts": {
    "test": "jest --runInBand --coverage=${CI}",
  }
}
```

Running `yarn test` results in:

```
Error: Unbound variable "CI"
    at interpolateArguments (/Users/brad/project/.yarn/releases/yarn-berry.js:10846:55)
    at executeCommandChain (/Users/brad/project/.yarn/releases/yarn-berry.js:10910:54)
    at executeCommandLine (/Users/brad/project/.yarn/releases/yarn-berry.js:10968:22)
    at executeCommandLine (/Users/brad/project/.yarn/releases/yarn-berry.js:10980:34)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

I'm migration from Yarn 1, and I would run this in CircleCI where the `CI` env var is defined and I use it to conditionally opt in to coverage. Calculating coverage is a little slower so I only do it by default in CI.

**Environment if relevant (please complete the following information):**

 - OS: OSX
 - Yarn version: 2

**Comment by arcanis** [comment]
Good find - it should be just a matter of adding an `else` here that would use `state.environment` before throwing the error:

https://github.com/yarnpkg/berry/blob/268d6a1c63e9bb743f5518552544e2af9020be1c/packages/berry-shell/sources/index.ts#L199

**Comment by arcanis** [comment]
This should be fixed starting from #225. Note that you'd still need to use `--coverage=${CI:-0}`, as otherwise you'll get the unbound variable error when the variable doesn't exist 🙂

**Comment by bradleyayers** [comment]
Perfect, thanks!
